Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -404,107 +352,72 @@ void CGPathAddArc(CGMutablePathRef path,
     }
 
     // Call the internal function for breaking the arcs down into smaller segments
-    _CGPathAddArc(path, m, x, y, radius, startAngle, endAngle, clockwise);
+    _CGPathAddArc(path, transform, x, y, radius, startAngle, endAngle, clockwise);
 }
 
 /**
  @Status Interoperable
 */
-void CGPathMoveToPoint(CGMutablePathRef path, const CGAffineTransform* m, float x, float y) {
-    if (path == NULL) {
-        return;
-    }
-
-    if (m) {
-        CGPoint pt;
-
-        pt.x = x;
-        pt.y = y;
-
-        pt = CGPointApplyAffineTransform(pt, *m);
-
-        x = pt.x;
-        y = pt.y;
+void CGPathMoveToPoint(CGMutablePathRef path, const CGAffineTransform* transform, float x, float y) {
+    CGPoint pt;
+    pt.x = x;
+    pt.y = y;
+    if (transform) {
+        pt = CGPointApplyAffineTransform(pt, *transform);
     }
-
-    _CGPathAddElement(path, kCGPathElementMoveToPoint, { x, y });
+    path->_impl.startingPoint = pt;
+    path->_impl.currentPoint = pt;
 }
 
 /**
  @Status Interoperable
 */
-void CGPathAddLines(CGMutablePathRef path, const CGAffineTransform* m, const CGPoint* points, size_t count) {
+void CGPathAddLines(CGMutablePathRef path, const CGAffineTransform* transform, const CGPoint* points, size_t count) {
     if (count == 0 || points == NULL || path == NULL) {
         return;
     }
 
-    CGPathMoveToPoint(path, m, points[0].x, points[0].y);
+    CGPathMoveToPoint(path, transform, points[0].x, points[0].y);
```

Comment:
>CGPathMoveToPoint [](start = 4, length = 17)

Is this correct? Move first, then add lines? Or does it add a line to the first point.